contracts/pcv/maple/MaplePCVDeposit.sol

```diff
@@ -0,0 +1,151 @@
+pragma solidity =0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {IPool} from "./IPool.sol";
+import {IMplRewards} from "./IMplRewards.sol";
+import {CoreRef} from "../../refs/CoreRef.sol";
+import {PCVDeposit} from "../PCVDeposit.sol";
+
+/// @notice PCV Deposit for Maple
+/// Allows depositing only by privileged role to prevent lockup period being extended by griefers
+
+/// Can only deposit USDC in this MAPLE PCV deposit
+contract MaplePCVDeposit is PCVDeposit {
+    using SafeERC20 for IERC20;
+
+    /// @notice reference to the Maple Pool where deposits and withdraws will originate
+    IPool public immutable pool;
+
+    /// @notice reference to the Maple Staking Rewards Contract
+    IMplRewards public immutable mplRewards;
+
+    /// @notice reference to the underlying token
+    IERC20 public immutable token;
+
+    /// @notice scaling factor for USDC
+    /// @dev hardcoded to use USDC decimals as this is the only
+    /// supplied asset Volt Protocol will support
+    uint256 public constant scalingFactor = 1e12;
+
+    /// @notice fetch underlying asset by calling pool and getting liquidity asset
+    /// @param _core reference to the Core contract
+    /// @param _pool reference to the Maple Pool contract
+    constructor(
+        address _core,
+        address _pool,
+        address _mplRewards
+    ) CoreRef(_core) {
+        token = IERC20(IPool(_pool).liquidityAsset());
+        /// enforce underlying token is USDC
+        require(
+            address(token) == 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48,
+            "MaplePCVDeposit: Underlying not USDC"
+        );
+        pool = IPool(_pool);
+        mplRewards = IMplRewards(_mplRewards);
+    }
+
+    /// @notice return the amount of funds this contract owns in Maple FDT's
+    /// without accounting for interest earned
+    /// does not account for unrealized losses in the venue
+    function balance() public view override returns (uint256) {
+        return pool.balanceOf(address(this)) / scalingFactor;
```

Comment:
add a test where we try different balance functions after advancing the time so that interest can accrue?

Reply:
advancing and then calling `accumulativeFundsOf` or `withdrawableFundsOf` doesn't work.

Reply:
`accumulativeFundsOf` returns interest accrued, `recognizableLossesOf` returns losses sustained, balanceOf returns amount of lp tokens.
summing accumulativeFundsOf and balanceOf, then subtracting recognizable losses returns the proper amount